### PR TITLE
Implemented Custom hook

### PR DESCRIPTION
**Pull request by NirzanMalla**
I have added custom hook called useFetch to avoid the repetition of logic for fetching api in every single pages throughout the app. I have also worked on the routing of home page. I have kept my custom hook in components directory.

### CODE REVIEW

File: src/components/UseFetch.jsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+import { useState, useEffect } from "react";
```
Comment (subhashishjs) [inline src/components/UseFetch.jsx]: always make a separate hook folder for this.

File: src/components/UseFetch.jsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import axios from "axios";
+
+const useFetch = (url) => {
+  const [data, setData] = useState([]);
```
Comment (subhashishjs) [inline src/components/UseFetch.jsx]: this is good. now we also want loading, error state in our hook.

File: src/pages/Cricket.jsx [hunk 3]
```diff
@@ -1,26 +1,12 @@
...
-      setData(response.data.articles);
-    } catch (error) {
-      console.error("Error fetching news:", error);
-    }
-  };
-
-  useEffect(() => {
-    getNews();
-  }, []);
+  const url = `https://newsapi.org/v2/everything?q=cricket&apiKey=${apikey}`;
```
Comment (subhashishjs) [inline src/pages/Cricket.jsx]: url nai kina pass garira. Only pass what is required. url lai ni keep it inside .env

File: src/pages/Football.jsx [hunk 4]
```diff
@@ -1,26 +1,12 @@
...
-      setData(response?.data?.articles);
-    } catch (error) {
-      console.error("Error fetching news:", error);
-    }
-  };
-
-  useEffect(() => {
-    getNews();
-  }, []);
+  const url = `https://newsapi.org/v2/everything?q=football&apiKey=${apikey}`;
```
Comment (subhashishjs) [inline src/pages/Football.jsx]: same, again code repetition. Always hate code repetition.

**Review by subhashishjs — COMMENTED** [review]
Agai code repetition cha. solve that!

File: src/components/UseFetch.jsx (hunk 1 above)
Comment (NirzanMalla) [reply src/components/UseFetch.jsx]: Done

File: src/components/UseFetch.jsx (hunk 2 above)
Comment (NirzanMalla) [reply src/components/UseFetch.jsx]: Done

File: src/pages/Cricket.jsx (hunk 3 above)
Comment (NirzanMalla) [reply src/pages/Cricket.jsx]: done

File: src/pages/Football.jsx (hunk 4 above)
Comment (NirzanMalla) [reply src/pages/Football.jsx]: done

**Comment by NirzanMalla** [comment]
I have resolved all the comments. I made different hook folder to keep my custom hook. I also added loading and error sate. I passed it by making a different component  to minimize code repetition. Further I worked on the way I passed direct urls, after consulting with other members, I decided to make utils folder where i created UrlHandler as a function and passed it on the pages changing only the required value. It not only helped me minimize code repetition but also made the code look clean.

**Review by subhashishjs — APPROVED** [review]

**Pull request merged**